### PR TITLE
Move worker build id to worker options, allow client id override there

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Moved (and renamed) worker build id to be on worker options, fairly obvious there.
Add ability to override client id from worker options too.

## Why?
Support https://github.com/temporalio/sdk-typescript/issues/697

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Review by cretz — APPROVED** [review]

File: client/src/lib.rs [hunk 1]
```diff
@@ -494,6 +494,17 @@ impl Client {
...
+    /// Return the options this client was initialized with mutably
+    pub fn options_mut(&mut self) -> &mut ClientOptions {
+        &mut self.inner.options
+    }
+
+    /// Set a worker build id to be attached to relevant requests. Unlikely to be useful outside
+    /// of core.
+    pub fn set_worker_build_id(&mut self, id: String) {
+        self.bound_worker_build_id = Some(id)
+    }
```
Comment (bergundy) [inline client/src/lib.rs:507]: Do you not plan on sharing this Client between different workers?

File: core/src/lib.rs [hunk 2]
```diff
@@ -68,11 +68,14 @@ where
...
     let as_enum = client.into();
-    // TODO: Assert namespaces match
     let client = match as_enum {
         AnyClient::HighLevel(ac) => ac,
         AnyClient::LowLevel(ll) => {
-            let client = Client::new(*ll, worker_config.namespace.clone());
+            let mut client = Client::new(*ll, worker_config.namespace.clone());
+            client.set_worker_build_id(worker_config.worker_build_id.clone());
+            if let Some(ref id_override) = worker_config.client_identity_override {
+                client.options_mut().identity = id_override.clone();
```
Comment (bergundy) [inline core/src/lib.rs:77]: Are the options here a clone of the original options that created the client?

**Review by bergundy — APPROVED** [review]

File: client/src/lib.rs (hunk 1 above)
Comment (Sushisource) [reply client/src/lib.rs:507]: All the important stuff is shared. Cloning them cones the underlying channel.

File: core/src/lib.rs (hunk 2 above)
Comment (Sushisource) [reply core/src/lib.rs:77]: Yes

File: core-api/src/worker.rs [hunk 3]
```diff
@@ -10,6 +10,14 @@ pub struct WorkerConfig {
     /// What task queue will this worker poll from? This task queue name will be used for both
     /// workflow and activity polling.
     pub task_queue: String,
+    /// A string that should be unique to the set of code this worker uses. IE: All the workflow,
+    /// activity, interceptor, and data converter code.
+    pub worker_build_id: String,
```
Comment (bergundy) [inline core-api/src/worker.rs:15]: We should have just called this `build_id`, I overlooked this in the review

**Pull request merged**